lib/src/main/java/io/ably/lib/realtime/Channel.java

```diff
@@ -491,6 +511,11 @@ private void failQueuedMessages(ErrorInfo reason) {
 	 * @throws AblyException
 	 */
 	public PaginatedResult<Message> history(Param[] params) throws AblyException {
+		/* We don't want to mess with users data. Acquire a copy of params array */
+		params = Arrays.copyOf(params, params.length);
```

Comment:
Instead of doing a copy and then replacing the param why not let replacePlaceholderParams return a new Params object as it is iterating over the Params anyway and could easily build a new Array.  One less step?